server/src/test/java/org/opensearch/index/shard/IndexShardTests.java

```diff
@@ -4036,8 +4036,7 @@ public void testCloseShardWhileResettingEngine() throws Exception {
         CountDownLatch closeDoneLatch = new CountDownLatch(1);
         IndexShard shard = newStartedShard(false, Settings.EMPTY, config -> new InternalEngine(config) {
             @Override
-            public InternalEngine recoverFromTranslog(TranslogRecoveryRunner translogRecoveryRunner, long recoverUpToSeqNo)
-                throws IOException {
+            public Engine recoverFromTranslog(TranslogRecoveryRunner translogRecoveryRunner, long recoverUpToSeqNo) throws IOException {
```

Comment:
Can we revert this change. Any reason why couldn't use `TranslogEventListener` here?

Reply:
Reverted the above change. Using `TranslogEventListener`, the test is getting stuck. Will debug it further.